Sources/SwiftOperatorPrecedence/OperatorPrecedence+Semantics.swift

```diff
@@ -0,0 +1,158 @@
+//===-------------- OperatorPrecedence+Semantics.swift --------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+import SwiftSyntax
+
+extension PrecedenceGroup {
+  /// Form the semantic definition of a precedence group given its syntax.
+  ///
+  /// TODO: This ignores all semantic errors.
+  init(from syntax: PrecedenceGroupDeclSyntax) {
+    self.name = syntax.identifier.text
+    self.syntax = syntax
+
+    for attr in syntax.groupAttributes {
+      // Relation (lowerThan, higherThan)
+      if let relation = attr.as(PrecedenceGroupRelationSyntax.self) {
+        let isLowerThan = relation.higherThanOrLowerThan.text == "lowerThan"
+        for otherGroup in relation.otherNames {
+          let otherGroupName = otherGroup.name.text
+          let relationKind: PrecedenceRelation.Kind = isLowerThan ? .lowerThan
+                                                                  : .higherThan
+          let relation = PrecedenceRelation(
+            kind: relationKind, groupName: otherGroupName, syntax: otherGroup)
+          self.relations.append(relation)
+        }
+
+        continue
+      }
+
+      // Assignment
+      if let assignment = attr.as(PrecedenceGroupAssignmentSyntax.self) {
+        self.assignment = assignment.flag.text == "true"
+        continue
+      }
+
+      // Associativity
+      if let associativity = attr.as(PrecedenceGroupAssociativitySyntax.self) {
+        switch associativity.value.text {
+        case "left":
+          self.associativity = .left
+
+        case "right":
+          self.associativity = .right
+
+        case "none":
+          self.associativity = .none
+
+        default:
+          break
+        }
+      }
+    }
+  }
+}
+
+extension Operator {
+  /// Form the semantic definition of an operator given its syntax.
+  ///
+  /// TODO: This ignores all semantic errors.
+  init(from syntax: OperatorDeclSyntax) {
+    self.syntax = syntax
+
+    let kindModifier = syntax.modifiers?.first { modifier in
+      OperatorKind(rawValue: modifier.name.text) != nil
+    }
```

Comment:
To avoid the force unwrap and `if` below, what do you think about
```suggestion
    kind = syntax.modifiers?.compactMap({ OperatorKind(rawValue: $0.name.text) }).first ?? .infix
```